conformance/lang/expressions/unary-expression/unary-minus-expression.balt

```diff
@@ -0,0 +1,984 @@
+
+function init() {
+    int a = -(let int h1 = 3 * 4 in h1);
+    io:println(a); // @output -12
+}
+
+Test-Case: output
+Description: Test multiplicative and additive expression in unary negation operands.
+Labels: additive-expr, DecimalNumber, int, multiplicative-expr, unary-minus
+
+function init() {
+    int a = -(1 + 49);
+    int b = -(23 * 4);
+    int c = -(100 / 4);
+    int d = -(99 % 5);
+    io:println(a); // @output -50
+    io:println(b); // @output -92
+    io:println(c); // @output -25
+    io:println(d); // @output -4
+}
+
+Test-Case: output
+Description: Test assignability result of unary negation expression to any.
+Labels: any, DecimalNumber, int, unary-minus
+
+function init() {
+    int a = 5;
+    any b = -a;
+    io:println(b == -5); // @output true
+}
+
+Test-Case: output
+Description: Test assignability result of unary negation expression to union type.
+Labels: int, DecimalNumber, string, unary-minus, union-type
+
+function init() {
+    int a = 5;
+    int|string b = -a;
+    io:println(b == -5); // @output true
+}
+
+Test-Case: output
+Description: Test unary expression as the operand of unary negation.
+Labels: DecimalNumber, int, unary-minus, unary-plus, unary-complement
+
+function init() {
+    int a = 5;
+    a = -+5;
+    io:println(a); // @output -5
+
+    a = 5;
+    a = --5;
+    io:println(a); // @output 5
+
+    a = 5;
+    a = -~5;
+    io:println(a); // @output 6
+}
+
+Test-Case: output
+Description: Test type cast expression in integer unary minus operands.
+Labels: DecimalNumber, int, type-cast-expr, unary-minus
+
+function init() {
+    int i = -<int> 6;
+    io:println(i); // @output -6
+}
+
+Test-Case: output
+Description: Test type cast expression in float unary minus operands.
+Labels: DecimalNumber, float, type-cast-expr, unary-minus
+
+function init() {
+    float i = -<float> 6;
+    io:println(i); // @output -6.0
+}
+
+Test-Case: output
+Description: Test shift expressions as unary negative operands.
+Labels: DecimalNumber, int, shift-expr, unary-minus
+
+function init() {
+    int p = -(1 << 2);
+    io:println(p); // @output -4
+}
+
+Test-Case: output
+Description: Test binary bitwise expressions as unary negative operands.
+Labels: binary-bitwise-expr, DecimalNumber, int, unary-minus
+
+function init() {
+    int a = -(2 & 6);
+    int b = -(2 ^ 6);
+    io:println(a); // @output -2
+    io:println(b); // @output -4
+}
+
+Test-Case: output
+Description: Test conditional expressions as unary negative operands.
+Labels: conditional-expr, DecimalNumber, error, int, union-type, unary-minus, optional-type
+
+function init() {
+    int|error a = 12;
+    int? b = ();
+    int c = -(a is error ? 2 : a);
+    int d =  -(b ?: 13);
+    io:println(c); // @output -12
+    io:println(d); // @output -13
+}
+
+Test-Case: output
+Description: Test checking expressions as unary negative operands.
+Labels: checking-expr, DecimalNumber, error, int, unary-minus, union-type
```

Comment:
Instead of `checking-expr`, we use `check` and `checkpanic` labels. Let's update where applicable.

Reply:
Updated in 79de921